### ISSUE TITLE
Location issues

### ISSUE THREAD

**Issue by ashinckley** [vejnoe/weather-in-your-calendar#14]
Lately I've been encountering an issue where my city will randomly come up as, "CITY NOT FOUND".  It was working earlier, and the last few times I've added this to my calendar, it works for about a day before showing up as not found.  One interesting behavior I noticed about the https://weather-in-calendar.com/ page is that after I added the city into the bar, it would find my city.  However, when I toggled the units or high/low buttons, it would sometimes change to "CITY NOT FOUND".  I can't seem to find a consistent pattern as to what "breaks" it.  
My city is Maple Valley, Washington.

**Comment by skylarp** [comment]
I'm having the same issue in Madison, Wisconsin.

**Comment by skylarp** [comment]
Okay, following the suggestion in [this comment on Issue 12](https://github.com/vejnoe/weather-in-your-calendar/issues/12#issuecomment-1850361892), I tried **Madison,WI,USA** as the input at weather-in-calendar.com and it is working fine. Fingers crossed that it will stay that way.

**Comment by livejamie** [comment]
Formatting like "Madison,WI,USA" does not fix the issue any longer

**Comment by livejamie** [comment]
I wonder if it's an API issue/limitation

**Comment by jengjo** [comment]
Does not work for me either. Oslo

**Comment by skylarp** [comment]
Yeah, my "fix" stopped working for me yesterday as well.

**Comment by philperrin** [comment]
Seeing that this code hasn't been updated in 4 years I don't know if a fix will come?
I assume it is related to the API call using v2.5, but the service is on v3.0. 
I have forked it to make an update - but could also be due to limitations on the new API version.

**Comment by livejamie** [comment]
I sent @vejnoe an email and got an autoresponse that he's out sick, so I'll wait for them to return.

In the meantime, there's https://www.webcal.guru/ and https://www.meteomatics.com/en/weather-calendar/ as alternatives

They're not as customizable but it's better than a broken feed

**Comment by fhunter** [comment]
> Seeing that this code hasn't been updated in 4 years I don't know if a fix will come? I originally posted about the api version perhaps not being correct, but that is not the case. The code in this repo uses: `http://api.openweathermap.org/data/2.5/forecast/daily?q=` But I found that using that with the free API does not work. However, using: `https://api.openweathermap.org/data/2.5/forecast?q=` Does work just fine with the free API. But I am not sure if this code will work for public use if everyone uses the same API key.

There was email from openweathermap, saying that they are deprecating 2.5 API in favor of 3.0 API (but as far as I looked, it was about the same - only URL was changing, for the most part)

**Comment by change-agent** [comment]
This keeps happening each week now

**Comment by livejamie** [comment]
It appears to be working again, weird.

**Comment by mojo2012** [comment]
webcal.guru did the trick for me :-)

**Comment by Killerbert** [comment]
Same problem here. I did add and remove the calendar multiple times this month.

Now I changed to meteomatics. So far so good

**Comment by livejamie** [comment]
Yep switched to Meteomatics as well, I hope OP's health issues aren't serious and they're doing ok.

**Comment by Killerbert** [comment]
Yea, hope OP is recovering well

**Comment by LucasLarson** [comment]
My short term solution was to enter my city name with the spaces pre-escaped:

Instead of `New York`, I entered `New%20York`, and the resulting URL has New York double escaped to `New%2520York`:
`webcal://weather-in-calendar.com/cal/weather-cal.php?city=New%2520York&units=imperial&temperature=day`

**Comment by lisaq** [comment]
Hi all. I took a look at the code, and it also accepts a zip code. I'm in Germany, but my Berlin zip code did work to fetch the correct location. In the meanwhile, it seems from @fhunter's comment that updating the openweathermap API would fix this, if someone wants to make a pull request or clone it. @vejnoe has not been active online for over a year, I hope he's ok.

**Comment by livejamie** [comment]
Hi Lisa, the script works intermittently. It will accept zip codes and other formats and stop working for a few days and come back.
